java/code/src/com/redhat/rhn/manager/system/VirtualInstanceManager.java

```diff
@@ -317,6 +374,27 @@ public static void updateGuestVirtualInstance(VirtualInstance virtualInstance,
     public static void updateGuestVirtualInstance(VirtualInstance virtualInstance,
```

Comment:
This method is called for each guest on a salt-based virtualization host, when the virtualization beacon fires. Doesn't this "truncate" `payg` flags  of all hosts to `false`?

Reply:
The PAYG attribute could only be true for Public Cloud instances (which can be PAYG or BYOS).

In case of a private cloud, all instances should be BYOS, therefore `payg` is set to `false`.